crates/chain-state/src/noop.rs

```diff
@@ -8,7 +8,11 @@ use reth_primitives_traits::NodePrimitives;
 use reth_storage_api::noop::NoopProvider;
 use tokio::sync::{broadcast, watch};
 
-impl<C: Send + Sync, N: NodePrimitives> CanonStateSubscriptions for NoopProvider<C, N> {
+impl<C, N> CanonStateSubscriptions for NoopProvider<C, N>
+where
+    C: Send + Sync,
+    N: NodePrimitives + Unpin + Default + Clone + 'static,
```

Comment:
places that require trait bounds like these can be collapsed into some helper trait eventually `HalfFullNodePrimitives` or smthg. `NodePrimitives` can't be static for example in order to be supertrait of traits of `T` that should be implemented on `&T` also. @mattsse